### PR TITLE
Support --configfile in NuGet CommandLine XPlat.

### PR DESCRIPTION
**Pull request by eerhardt**
Adding a --configfile option to nuget command line in order to support "dotnet restore" taking a config file.

Fix https://github.com/NuGet/Home/issues/2033

@yishaigalatzer, @anurse  Please let me know what you think.

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
Looks good to me, but needs a test.

CC @emgarten

**Comment by eerhardt** [comment]
I'd love to add a test.  Can you point me to the existing tests for CommandLine.XPlat?

**Comment by yishaigalatzer** [comment]
NuGet.CommandLine.Xplat.Tests ?

**Comment by eerhardt** [comment]
Thanks, @yishaigalatzer.  I added the test and squashed the commits.  Please take a look and let me know what you think.

File: test/NuGet.Core.Tests/NuGet.CommandLine.XPlat.Test/BasicLoggingTests.cs [hunk 1]
```diff
@@ -69,6 +69,29 @@ public void BasicLogging_RestoreHelp_ExitCode()
         }
 
         [Fact]
+        public void BasicLogging_RestoreConfigFile_ExitCode()
+        {
+            // Arrange
+            var log = new TestLogger();
```
Comment (yishaigalatzer) [inline test/NuGet.Core.Tests/NuGet.CommandLine.XPlat.Test/BasicLoggingTests.cs]: I would expect a test using the custom config file rather than just the negative test

Comment (eerhardt) [inline test/NuGet.Core.Tests/NuGet.CommandLine.XPlat.Test/BasicLoggingTests.cs]: I'm not sure how to exactly do this.  There isn't an existing test that actually DOES a restore.  All the tests in this project either:
1. are negative tests
2. Check that "restore --help" returns 0 exit code
3. empty args returns 0 exit code.

Comment (eerhardt) [inline test/NuGet.Core.Tests/NuGet.CommandLine.XPlat.Test/BasicLoggingTests.cs]: In my mind, this test does more than just testing the negative.  It:
1. Ensures the "--configfile" doesn't fail because it is an unrecognized option.
2. NuGet is at least attempting to look for the file passed in with --configfile.

Comment (emgarten) [inline test/NuGet.Core.Tests/NuGet.CommandLine.XPlat.Test/BasicLoggingTests.cs]: There isn't a way to verify the restore parameters from xplat until after moving from dnx to dotnet cli. 

XPlat loads an assembly from DNX RC2, however the tests run under DNX RC1. When the restore method is hit a missing method exception is thrown.

Restore is currently tested at the RestoreCommand layer.

Comment (yishaigalatzer) [inline test/NuGet.Core.Tests/NuGet.CommandLine.XPlat.Test/BasicLoggingTests.cs]: OK. Lets file a follow up issue after moving to .net CLI.

Comment (eerhardt) [inline test/NuGet.Core.Tests/NuGet.CommandLine.XPlat.Test/BasicLoggingTests.cs]: Logged https://github.com/NuGet/Home/issues/2040 for this issue.

Is this PR ok to merge?

**Comment by emgarten** [comment]
@eerhardt would you rebase this against dev? I'm getting conflicts. After that I think this change should go in.

**Comment by eerhardt** [comment]
Thanks, @emgarten.  I've rebased and pushed the latest.

**Comment by emgarten** [comment]
Merged with NuGet/NuGet.Client@cf8f76c2e5e8b41edb61661e9a5b856a146fa50a

**Comment by emgarten** [comment]
Thanks for the PR @eerhardt
